test/server/cards/01_SOR/upgrades/SnapshotReflexes.spec.ts

```diff
@@ -0,0 +1,45 @@
+describe('Snapshot Reflexes', function() {
+    integration(function(contextRef) {
+        describe('Snapshot Reflexes\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['battlefield-marine'],
+                        hand: ['snapshot-reflexes']
+                    },
+                    player2: {
+                        groundArena: ['specforce-soldier']
+                    }
+                });
+            });
+
+            it('attached unit may attack when played', function () {
```

Comment:
The string argument for each `it` should form a complete sentence when added to the string from the `describe` call above. That sentence is output when a test fails, so it should describe what the test is supposed to prove so that whoever reads it gets an idea of what failed.

Reply:
+1. If you want to see how the test name looks, you can just force the test to fail and it will print it out in the error message

Reply:
I modified the texts to be more self descriptive and describe the upgrade behaviour